### PR TITLE
Describe 4D arrays in Background and Concepts docs

### PR DESCRIPTION
**Pull request by rafmudaf**
# Describe 4D data structures in Background and Concepts notebook
The [Background and Concepts](https://nrel.github.io/floris/floris_101.html) page in the documentation is meant to be a resource for new users to understand the critical ideas in FLORIS. One important concept is the dimensionality of the data structures. This was initially added during the v3 development work, and I've updated it here to reflect the new structure of the arrays.

Additionally, there are a few minor fixes included in this pull request.

## Additional supporting information
Since this pull request mostly deals with changing a Jupyter Notebook and those are difficult to review on GitHub, I suggest to build the documentation locally and suggest any changes in a comment.

@misi9170 @paulf81

### CODE REVIEW

File: docs/floris_101.ipynb [hunk 1]
```diff
@@ -192,9 +196,9 @@
      "output_type": "stream",
      "text": [
       "Dimensions of grid x-components\n",
-      "(2, 2, 4, 3, 3)\n",
+      "(4, 4, 3, 3)\n",
       "\n",
-      "Turbine 3 grid x-components for first wind direction and first wind speed\n",
+      "3rd turbine x-components for first wind condition\n",
```
Comment (paulf81) [inline docs/floris_101.ipynb]: Could say "(findex)" after "first wind condition"

File: docs/floris_101.ipynb [hunk 2]
```diff
@@ -129,11 +129,12 @@
    "metadata": {},
    "source": [
     "Additionally, we can change the wind speeds and wind directions.\n",
-    "These are lists of wind speeds and wind directions that will be\n",
-    "combined so that a wake calculation will happen for every wind\n",
-    "direction with each speed.\n",
+    "The set of wind conditions is given as an array of combinations\n",
```
Comment (paulf81) [inline docs/floris_101.ipynb]: "Combinations" could be a confusing word here as it implies FLORIS will assemble all combinations, as in V3, maybe could say "conditions".  Could be useful to say, this is similar to the "time_series" mode in v3

**Comment by paulf81** [comment]
That's looking good, I left only very small comments, thanks @rafmudaf

File: docs/floris_101.ipynb (hunk 1 above)
Comment (rafmudaf) [reply docs/floris_101.ipynb]: Good idea, I added `(at findex=0)`

File: docs/floris_101.ipynb (hunk 2 above)
Comment (rafmudaf) [reply docs/floris_101.ipynb]: Yes, good point. I changed this to:
```
Additionally, we can change the wind speeds and wind directions.
The set of wind conditions is given as arrays of wind speeds and
wind directions that combined describe the atmospheric conditions
to compute. This requires that the wind speed and wind direction
arrays be the same length.
```
And I added more information in the code block below it:
```python
# One wind direction and one speed
# -> one atmospheric condition (270 degrees at 8 m/s)
fi.reinitialize(wind_directions=[270.0], wind_speeds=[8.0])

# Two wind directions and one speed (repeated)
# -> two atmospheric conditions (270 degrees at 8 m/s and 280 degrees at 8 m/s)
fi.reinitialize(wind_directions=[270.0, 280.0], wind_speeds=[8.0, 8.0])

# Two wind directions and two speeds combined
# -> four atmospheric conditions (270 degrees at 8 m/s and 9 m/s, 280 degrees at 8 m/s and 9 m/s)
fi.reinitialize(wind_directions=[270.0, 280.0, 270.0, 280.0], wind_speeds=[8.0, 8.0, 9.0, 9.0])
```

I'd like this page to be kind of a "getting started" guide and referring to v3 implies that the reader has some knowledge of FLORIS already. I left that part out, but I think a transition guide would be very helpful, and it should include what you noted about "time_series" mode in v3 being the standard in v4.

**Pull request merged**